src/inausoft.netCLI.Tests/Executor.Tests.cs

```diff
@@ -101,28 +101,28 @@ public void Executor_RunCLI_ShouldRunOnlyProperCommandHandler_WhenMultipleComman
             Assert.AreEqual(0, mockCommand1Handler.LastRunParameters.IntOption);
         }
 
-
-        [ExpectedException(typeof(InvalidCommandException))]
         [TestMethod]
         public void Executor_RunCLI_ShouldThrowInvalidCommmandException_ForNotRegistratedCommand()
         {
+            var invalidCommandName = "invalidCommandName";
+
             //Arrange
             var args = new string[]
                             {
-                                "commandXX",
+                                invalidCommandName,
                                 "--boolOption", "true",
                             };
 
             var mockCommand1Handler = new MockCommand1Handler();
             var config = new CLIConfiguration().Map<Command1, MockCommand1Handler>();
 
             //Act
-            Executor.RunCLI(config, args, mockCommand1Handler, mockCommand1Handler);
+            var ex = Assert.ThrowsException<InvalidCommandException>(() => Executor.RunCLI(config, args, mockCommand1Handler));
 
-            //Assert with exception
+            //Assert
+            Assert.AreEqual(invalidCommandName, ex.CommandName);
         }
 
-        [ExpectedException(typeof(InvalidCommandException))]
         [TestMethod]
         public void Executor_RunCLI_ShouldThrowInvalidCommmandException_ForEmptyArguments()
```

Comment:
This actually not an "invalid command", it's more like "no command at all". Have you considered introducing concept similar to `CommandMissingException`?

Reply:
I will add that.

Reply:
I would left it for now. And would come back to it during #9. I guess we could throw `InvalidOperationEception` there as since we would have default command this situation would be much more rare.